src/org/mangui/hls/controller/BufferThresholdController.as

```diff
@@ -1,80 +1,84 @@
-            var metrics : HLSLoadMetrics = event.loadMetrics;
-            // only monitor main fragment metrics for buffer threshold computing
-            if(metrics.type == HLSLoaderTypes.FRAGMENT_MAIN) {
-                /* set min buf len to be the time to process a complete segment, using current processing rate */
-                _minBufferLength = metrics.processing_duration * (_targetduration / metrics.duration);
-                // avoid min > max
-                if (HLSSettings.maxBufferLength) {
-                    _minBufferLength = Math.min(HLSSettings.maxBufferLength, _minBufferLength);
-                }
-
-                // avoid _minBufferLength > minBufferLengthCapping
-                if (HLSSettings.minBufferLengthCapping > 0) {
-                    _minBufferLength = Math.min(HLSSettings.minBufferLengthCapping, _minBufferLength);
-                }
-
-                CONFIG::LOGGING {
-                    Log.debug2("AutoBufferController:minBufferLength:" + _minBufferLength);
-                }
-            };
-        }
-    }
-}
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package org.mangui.hls.controller {
+    import org.mangui.hls.constant.HLSLoaderTypes;
+    import org.mangui.hls.event.HLSEvent;
+    import org.mangui.hls.event.HLSLoadMetrics;
+    import org.mangui.hls.HLS;
+    import org.mangui.hls.HLSSettings;
+
+    CONFIG::LOGGING {
+        import org.mangui.hls.utils.Log;
+    }
+    /** Class that manages buffer threshold values (minBufferLength/lowBufferLength)
+     */
+    public class BufferThresholdController {
+        /** Reference to the HLS controller. **/
+        private var _hls : HLS;
+        private var _targetduration : Number;
+        private var _minBufferLength : Number;
+        private var _minLiveBufferLength : Number;
+
+        /** Create the loader. **/
+        public function BufferThresholdController(hls : HLS) : void {
+            _hls = hls;
+            _hls.addEventListener(HLSEvent.MANIFEST_LOADED, _manifestLoadedHandler);
+            _hls.addEventListener(HLSEvent.TAGS_LOADED, _fragmentLoadedHandler);
+            _hls.addEventListener(HLSEvent.FRAGMENT_LOADED, _fragmentLoadedHandler);
+        };
+
+        public function dispose() : void {
+            _hls.removeEventListener(HLSEvent.MANIFEST_LOADED, _manifestLoadedHandler);
+            _hls.removeEventListener(HLSEvent.TAGS_LOADED, _fragmentLoadedHandler);
+            _hls.removeEventListener(HLSEvent.FRAGMENT_LOADED, _fragmentLoadedHandler);
+        }
+
+        public function get minBufferLength() : Number {
+            var mbl : Number = HLSSettings.minBufferLength == -1
+				? _minBufferLength
+				: HLSSettings.minBufferLength;
+
+			return _hls.isAltAudio
+				? Math.max(mbl, _minLiveBufferLength)

```

Comment:
should this be the `max` or the `min`?

Reply:
So... this is my theory: Playback was stalling because the buffer wasn't big enough to account for the encrypted content, I think this allows for a larger buffer (so we would want the larger of the two) to account for the size of the encrypted segments.

Reply:
🤔 ok, that makes logical sense to me